### PR TITLE
Support grid styles using CSS variables

### LINKED ISSUE #50946: Grid: support sash theming

### PR DESCRIPTION
**Pull request by joaomoreno**
Fixes #50946

### CODE REVIEW

**Comment by bpasero** [comment]
:cheers:

File: src/vs/base/browser/ui/splitview/splitview.ts [hunk 1]
```diff
@@ -137,6 +147,18 @@ export class SplitView implements IDisposable {
...
+		this.style(options.styles || defaultStyles);
+	}
+
+	style(styles: ISplitViewStyles): void {
+		if (styles.separatorBorder.isTransparent()) {
+			dom.removeClass(this.el, 'separator-border');
+			this.el.style.removeProperty('--separator-border');
+		} else {
+			dom.addClass(this.el, 'separator-border');
+			this.el.style.setProperty('--separator-border', styles.separatorBorder.toString());
```
[the remark this answers is not in the document]
Comment (chrmarti) [reply src/vs/base/browser/ui/splitview/splitview.ts]: @joaomoreno I guess these are scoped to the element and inherited by its child elements?

[the remark this answers is not in the document]
Comment (bpasero) [reply src/vs/base/browser/ui/splitview/splitview.ts]: > This is probably the way to go for instance specific theming: button, tree, etc...

Not as long we have to support browsers for the standalone editor that do not support CSS variables...

[the remark this answers is not in the document]
Comment (joaomoreno) [reply src/vs/base/browser/ui/splitview/splitview.ts]: @chrmarti Correct.

@bpasero Also correct, forgot about that.

**Pull request merged**